src/modules/src/Eryph.Modules.ComputeApi/Handlers/GetCatletConfigurationHandler.cs

```diff
@@ -76,9 +76,21 @@ public GetCatletConfigurationHandler(IStateStore stateStore)
 
                 if (drive.AttachedDisk != null)
                 {
-                    driveConfig.Name = Path.GetFileNameWithoutExtension(drive.AttachedDisk.FileName);
-                    driveConfig.Size = (int)
-                        Math.Ceiling(drive.AttachedDisk.SizeBytes.GetValueOrDefault() / 1024d / 1024 / 1024);
+                    driveConfig.Name = drive.AttachedDisk.Name;
+
+                    driveConfig.Size = (int)Math.Ceiling(
+                        drive.AttachedDisk.SizeBytes.GetValueOrDefault() / 1024d / 1024 / 1024);
+
+                    // When the drive's parent is a gene disk and their sizes are equal,
+                    // this means the drive size has not been changed and should be omitted.
+                    // This check handles the case when the catlet gene in the gene pool has no
+                    // disk sizes specified in the config.
+                    if (drive.AttachedDisk.Parent is not null
+                        && drive.AttachedDisk.Parent.StorageIdentifier?.StartsWith("gene:") == true
```

Comment:
mmhm, could this be simplified to be without gene check? Thinking about multiple manually created parent/child relationships with a volume that is shared as a parent without being a gene.

Reply:
discussed, ok